scipy/special/_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,75 @@
+import os
+import sys
+import functools
+
+import numpy as np
+from scipy._lib._array_api import array_namespace, is_cupy, is_torch, is_numpy
+from . import _ufuncs
+# These don't really need to be imported, but otherwise IDEs might not realize
+# that these are defined in this file / report an error in __init__.py
+from ._ufuncs import (
+    log_ndtr, ndtr, ndtri, erf, erfc, i0, i0e, i1, i1e,  # noqa
+    gammaln, gammainc, gammaincc, logit, expit)  # noqa
+
+_SCIPY_ARRAY_API = os.environ.get("SCIPY_ARRAY_API", False)
```

Comment:
I'm not sure it really makes any difference, but why not import this from `scipy._lib._array_api`?

```python

from scipy._lib._array_api import SCIPY_ARRAY_API
```

Also, it looks like `SCIPY_ARRAY_API` can  be a json string giving a list of backends to support. See https://github.com/scipy/scipy/blob/7e96eff33bb864a6413ea1475b3b1cb486e652b5/scipy/conftest.py#L126-L138

I'm not completely up to speed on this stuff, and don't know the design decisions involved, but should we also offer that level of granularity for the dispatch in special?

Reply:
Support for the list is only intended to be for testing. Otherwise we only want to tell users to activate or not the feature.